src/main/ruby/truffleruby/core/struct.rb

```diff
@@ -56,7 +56,7 @@ def self.new(klass_name, *attrs, keyword_init: nil, &block)
     end
 
     klass = Class.new self do
-      _specialize attrs unless keyword_init
+      _specialize attrs unless Primitive.nil?(keyword_init) || keyword_init
```

Comment:
```suggestion
      _specialize attrs if Primitive.false?(keyword_init)
```

Reply:
Will fix it.

Reply:
Thanks.  I knew there wasn't type checking being done, but that would only matters if checking `.true?` 🤦